modules/boltlib/lib/puppet/functions/file_upload.rb

```diff
@@ -0,0 +1,53 @@
+# Uploads the given file or directory to the given set of targets and returns the result from each upload.
+#
+# * This function does nothing if the list of targets is empty.
+# * It is possible to run on the target 'localhost'
+# * A target is a String with a targets's hostname or a Target.
+# * The returned value contains information about the result per target.
+#
+# Since > 5.4.0 TODO: Update when version is known
+#
+Puppet::Functions.create_function(:file_upload, Puppet::Functions::InternalFunction) do
+  local_types do
+    type 'TargetOrTargets = Variant[String[1], Target, Array[TargetOrTargets]]'
```

Comment:
Where does the Target type come from? It doesn't appear to be setup during testing, but I'm not even finding where it's defined.